pkgs/desktops/lomiri/development/lomiri-ui-toolkit/default.nix

```diff
@@ -0,0 +1,312 @@
+      "tst_slider.11.qml"
+      "tst_slider.13.qml"
+      "tst_switch_bug1510919.13.qml"
+      "tst_tabs.11.qml"
+      "tst_tabs.13.qml"
+      "tst_tabs.DEPRECATED_TOOLBAR.11.qml"
+      "tst_textarea.11.qml"
+      "tst_textarea_in_flickable.11.qml"
+      "tst_textfield.11.qml"
+      "tst_textinput_common.12.qml"
+      "tst_textinput_common.13.qml"
+      "tst_toggles.13.qml"
+
+      # SignalSpy on QML shader compilers are failing for unknown reasons
+      "tst_icon.11.qml"
+      "tst_icon.13.qml"
+    ]}; do
+      find tests/unit -name $badQmlTest -delete
+    done
+
+    # These tests are registered explicitly, just removing them from the list is enough
+    for badUnitTest in ${lib.strings.concatStringsSep " " [
+      # Requires a working Qt OpenGL context in ShapeMaterial
+      "inversemousearea"
+      "layouts"
+      "recreateview"
+      "subtheming"
+      "swipearea"
+    ]}; do
+      sed -i -e "/$badUnitTest/d" tests/unit/unit.pro
+    done
+
+    # These tests don't fail explicitly, but issue warnings that the test suite doesn't expect, which it turns into errors
+    for unexpectedWarningTest in ${lib.strings.concatStringsSep "" [
+      # qml/Lomiri/Components/1.2/Toolbar.qml:96: TypeError: Cannot read property 'locked' of null
+      # I don't know QML enough to fix this, upstream supposedly knows about this already
+      "tst_pagestack.DEPRECATED_TOOLBAR.11.qml"
+    ]}; do
+      sed -i -e "/EXCEPTIONS=/a '$unexpectedWarningTest' \\" tests/checkresults.sh
+    done
+
+    # Override the overriding of regular Qt variables for tests
+    substituteInPlace export_qml_dir.sh \
+      --replace '_IMPORT_PATH=$BUILD_DIR/qml' '_IMPORT_PATH=$BUILD_DIR/qml:${qtQmlPaths}' \
+      --replace 'LOMIRI_UI_TOOLKIT_THEMES_PATH=$BUILD_DIR/qml' 'LOMIRI_UI_TOOLKIT_THEMES_PATH=$BUILD_DIR/qml:${ubuntu-themes}/share'
+
+    # This test overrides XDG_RUNTIME_DIR with a custom path that it manages, in an attempt to ensure correct permissions
+    # Ironically this causes a warning by Qt about the permissions being wrong (755 instead of 700), which the test suite treats as an error
+    sed -i \
+      -e '/setenv("XDG_RUNTIME_DIR", testRuntimeDir.toUtf8()/a QFile::setPermissions(testRuntimeDir.toUtf8(), QFileDevice::ReadOwner | QFileDevice::WriteOwner | QFileDevice::ExeOwner);' \
+      tests/unit/statesaver/tst_statesaver.cpp
+  '';
+
+  nativeBuildInputs = [
+    perl
+    pkg-config
+    python3
+    qmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    glib
+    lttng-ust
+    qtbase
+    qtdeclarative
+    qtpim
+    qtquickcontrols2
+    qtsystems
+  ];
+
+  propagatedBuildInputs = [
+    qtfeedback
+    qtgraphicaleffects
+    qtsvg
+  ];
+
+  nativeCheckInputs = [
+    dbus-test-runner
+    dpkg # `dpkg-architecture -qDEB_HOST_ARCH` response decides how tests are run
+    gdb
+    xvfb-run
+  ];
+
+  dontWrapQtApps = true;
+
+  qmakeFlags = [
+    # docs require Qt's qdoc, which we don't have(?)
```

Comment:
clang-based c++ parser for qdoc and lupdate is disable by default,  https://github.com/NixOS/nixpkgs/issues/239487
Use `qttools.override { withClang = true; }`  to get it

Reply:
AFAICT #239781 only touched Qt6, this is Qt5.

https://github.com/wineee/nixpkgs/blob/da5286cd2b23e080b383594d6069e4487ef233de/pkgs/development/libraries/qt-5/modules/qttools.nix
https://github.com/wineee/nixpkgs/blob/da5286cd2b23e080b383594d6069e4487ef233de/pkgs/development/libraries/qt-6/modules/qttools.nix

Reply:
#245379 for the change to `libsForQt5.qttools` against staging. Docs aren't urgent for this so it's not a blocker for me, but I do have a docs-enabling commit ready.